source/_integrations/template.markdown

```diff
@@ -561,6 +561,28 @@ template:
 
 {% endraw %}
 
+A more advanced use case could be to set the icon based on the sensor's own state like above, but when triggered by an event. This example demonstrates a binary sensor that turns on momentarily, such as when a doorbell button is pressed. When the button gets pressed, the first trigger will be triggered and the binary sensor's state will be updated. This in turn will trigger the second trigger, which will cause the icon template to be evaluated. The platform check in the state template is to avoid the update from going into an infinite loop of updating. 
```

Comment:
This needs to be adjusted to the new situation :)

Reply:
Bussy week... :-) 

I updated it now. 

I dropped the infinite loop part, because it doesn't seem relevant at all now.